src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1219,6 +1222,52 @@ void OrbitMainWindow::on_actionOpen_Capture_triggered() {
   QProcess::startDetached(orbit_executable, arguments << file << command_line_flags_);
 }
 
+void OrbitMainWindow::on_actionRename_Capture_File_triggered() {
+  CHECK(target_label_->GetFilePath().has_value());
+  std::filesystem::path current_file_path = target_label_->GetFilePath().value();
+  std::filesystem::path file_name = current_file_path.filename();
+
+  auto* dialog = new QInputDialog(this);
+  dialog->setWindowTitle(tr("Rename Capture File"));
+  dialog->setLabelText(tr("File Name"));
+  dialog->setInputMode(QInputDialog::TextInput);
+  dialog->setTextEchoMode(QLineEdit::Normal);
+  dialog->setTextValue(QString::fromStdString(file_name.string()));
+  dialog->resize(600, dialog->height());
+
+  if (dialog->exec() == QDialog::Rejected) return;
+
+  std::string new_file_name = dialog->textValue().toStdString();
+
+  if (new_file_name == file_name.string()) return;
+
+  if (new_file_name.empty() || absl::StrContains(new_file_name, "/") ||
+      absl::StrContains(new_file_name, "\\")) {
+    QMessageBox::critical(
+        this, QString::fromStdString("Invalid File Name"),
+        QString::fromStdString(absl::StrFormat("Invalid file name: \"%s\"", new_file_name)));
+    return;
+  }
+
+  std::filesystem::path new_file_path = current_file_path.parent_path() / new_file_name;
+
+  ErrorMessageOr<void> rename_result = orbit_base::MoveFile(current_file_path, new_file_path);
```

Comment:
What happens if the new filename already exists?

Reply:
right. This I need to fix, thanks for catching it!

Reply:
done.